decred_wallet/Features/Wallet Setup/RecoverExistingWalletViewController.swift

```diff
@@ -102,65 +112,59 @@ class RecoverExistingWalletViewController: WalletSetupBaseViewController, UITabl
+            self.tableViewFooterTopSpacingConstraint.constant = 30
+            UIView.animate(withDuration: 0.5) {
+                self.tableViewFooter.layoutIfNeeded()
+            }
+        }
+        
         if wordIndex < 32 && moveToNextField {
             self.focusSeedWordInput(at: wordIndex + 1)
         } else {
             self.view.endEditing(true)
         }
         
         if self.validateSeed().valid {
-            self.activateConfirmButton()
+            self.btnConfirm.backgroundColor = UIColor.appColors.decredGreen
         } else {
-            self.deactivateConfirmButton()
+            self.btnConfirm.backgroundColor = UIColor.appColors.lightGray
         }
     }
     
     func focusSeedWordInput(at tableRowIndex: Int) {
         let tableIndexPath = IndexPath(row: tableRowIndex, section: 0)
-        
         let nextSeedWordCell = self.tableView.cellForRow(at: tableIndexPath) as? RecoveryWalletSeedWordCell
         nextSeedWordCell?.seedWordAutoComplete.becomeFirstResponder()
-        
-        self.tableView.scrollToRow(at: tableIndexPath, at: .middle, animated: true)
+    }
+
+    @IBAction func backButtonTap(_ sender: Any) {
+        self.navigationController?.popViewController(animated: true)
     }
     
-    func activateConfirmButton() {
-        self.btnConfirm.backgroundColor = UIColor.appColors.decredGreen
-        self.lblEnterAllSeeds.isHidden = true
-        
-        // increase top spacing since warning label is now hidden so as to position button in center
-        self.tableViewFooterTopSpacingConstraint.constant = 30
-        UIView.animate(withDuration: 0.5) {
-            self.tableViewFooter.layoutIfNeeded()
+    @IBAction func onConfirm() {
+        if self.userEnteredSeedWords.contains("") {
+            self.displaySeedError("Not all seeds are entered. Please, check input fields and enter all seeds.")
+        }
+        else{
+            let validatedSeed = self.validateSeed()
+            if validatedSeed.valid {
+                self.secureWallet(validatedSeed.seed)
+            }
+            else {
+                self.displaySeedError("You entered an incorrect seed. Please check your words.")
+            }
         }
     }
     
-    func deactivateConfirmButton() {
-        self.btnConfirm.backgroundColor = UIColor.appColors.lightGray
+    func displaySeedError(_ errorMessage: String) {
         self.lblEnterAllSeeds.isHidden = false
         
-        if self.userEnteredSeedWords.contains("") {
-            self.lblEnterAllSeeds.text = "Not all seeds are entered. Please, check input fields and enter all seeds."
-        } else {
-            self.lblEnterAllSeeds.text = "You entered an incorrect seed. Please check your words."
-        }
-        
         // reduce top spacing so that warning label and confirm button are centered in display
         self.tableViewFooterTopSpacingConstraint.constant = 10
-        UIView.animate(withDuration: 0.5) {
```

Comment:
This is required when changing the `tableViewFooterTopSpacingConstraint`. Please restore it.